share/blob.go

```diff
@@ -25,6 +25,9 @@ func NewBlob(ns Namespace, data []byte, shareVersion uint8, signer []byte) (*Blo
 	if len(data) == 0 {
 		return nil, errors.New("data can not be empty")
 	}
+	if ns.IsEmpty() {
```

Comment:
Now that we are enhancing the checks for namespace, wondering shouldn't we consider checking that namespace version is `0` as per [specs](https://celestiaorg.github.io/celestia-app/specs/namespace.html#version-0) (it seems there is no other supported versions) or otherwise the namespace is one of the [reserved namespaces](https://celestiaorg.github.io/celestia-app/specs/namespace.html#reserved-namespaces)?

Reply:
Makes sense. Then we don't have to remember to do it outside this function